### PR TITLE
Add Falling Block Expression

### PR DESCRIPTION
**Pull request by UnderscoreTud**
### Description
This PR adds a new expression for getting a falling block entity data using item types, blocks and block datas

---
**Target Minecraft Versions:** any
**Requirements:** none
**Related Issues:** https://github.com/SkriptLang/Skript/issues/4560

### CODE REVIEW

**Comment by TheLimeGlass** [comment]
Should be using EntityData and the lang file. Not a new expression.

**Review by TheLimeGlass — CHANGES_REQUESTED** [review]
I think `falling (block(|1¦s)|(2¦)%-itemtype%)` should support the word `of` also can this not be `%-itemtype/blockdata%`?

File: src/main/java/ch/njol/skript/entity/FallingBlockData.java [hunk 1]
```diff
@@ -122,75 +129,71 @@ public FallingBlock spawn(Location loc, @Nullable Consumer<FallingBlock> consume
...
 	}
-	
+
 	@Override
-	protected boolean equals_i(final EntityData<?> obj) {
+	protected boolean equals_i(EntityData<?> obj) {
 		if (!(obj instanceof FallingBlockData))
 			return false;
+		if (data != null && ((FallingBlockData) obj).data != null)
+			return data.equals(((FallingBlockData) obj).data);
```
Comment (AyhamAl-Ali) [inline src/main/java/ch/njol/skript/entity/FallingBlockData.java:190]: Can block data checking replace item type checking? if not it shouldn't return if true only if false

File: src/main/java/ch/njol/skript/entity/FallingBlockData.java [hunk 2]
```diff
@@ -122,75 +129,71 @@ public FallingBlock spawn(Location loc, @Nullable Consumer<FallingBlock> consume
...
 	@Override
 	public EntityData getSuperType() {
 		return new FallingBlockData(types);
 	}
-	
+
 	@Override
-	public String toString(final int flags) {
-		final ItemType[] types = this.types;
+	public String toString(int flags) {
```
Comment (AyhamAl-Ali) [inline src/main/java/ch/njol/skript/entity/FallingBlockData.java:168]: Does it support the newly added feature `data`?

**Review by AyhamAl-Ali — CHANGES_REQUESTED** [review]
Nice PR ⚡

**Comment by UnderscoreTud** [comment]
Not working on this anymore

**Pull request closed without merging**